### DISCUSSION TITLE
How to specify resize rules for Media Library

### DISCUSSION THREAD

**Discussion by omaksi** [strapi/strapi#7625] [Legacy Topics]
Hello,

I don't see any way to specify resize rules for the Media Library. I would like to specify custom resolutions and/or rules, like contain/cover, that would get applied automatically on image upload. Is this currently possible?

Thanks

**Comment by derrickmehaffy** [comment]
Currently I'm not sure if that is possible, I will ping @soupette and @HichamELBSI about this to see if they have any thoughts, but in this case I believe it would be best to create a feature request  issue :)

**Comment by lauriejim** [comment]
Hello! All image manipulation functions are located here - https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-upload/services/image-manipulation.js
By using the [customization concept](https://strapi.io/documentation/v3.x/concepts/customization.html) you will be able to update the function you want to add your custom code.

**Reply by omaksi** [reply]
Thanks @lauriejim 

Any chance of extracting the breakpoints and possibly other config to a config file? Overriding methods seems a bit risky with regards to updates and maintenance.

**Reply by lauriejim** [reply]
Right now there is no other way to do.
We are working on the internal API to updated this kind of thing.
